tools/pythonpkg/src/native/python_conversion.cpp

```diff
@@ -606,7 +633,7 @@ Value TransformPythonValue(py::handle ele, const LogicalType &target_type, bool
 		PyDictionary dict = PyDictionary(py::reinterpret_borrow<py::object>(ele));
 		switch (target_type.id()) {
 		case LogicalTypeId::STRUCT:
-			return TransformDictionaryToStruct(dict, target_type);
+			return TransformDictionaryToStruct(dict, target_type).DefaultCastAs(target_type);
```

Comment:
As we are already doing the re-ordering in the dictionary conversion, do we need to do the cast here as well? Can we perhaps construct the struct in the correct order directly in `TransformDictionaryToStruct`?

Reply:
We're not doing the reordering yet, we're only matching the key to the right target type.
I left reordering as a FIXME and let DefaultCastAs deal with it for now

Reply:
The re-ordering seems relatively simple no? You already get the target index from `key_mapping[struct_keys[i]]`, all that would be required is resizing the `struct_values` array to `dict.len`, and then assigning to the target index instead of emplacing.

It's probably not performance sensitive code but casting is adding extra cost as we will be binding the cast for every value, in addition to doing the key look-ups multiple times.

Reply:
Yea you're right, I think it will be trivial I just wanted to put it off for later, but I'll address the FIXME 👍

Reply:
Fixed it